src/manager.cpp

```diff
@@ -199,6 +224,159 @@ sdbusplus::async::task<bool>
     co_return true;
 }
 
+sdbusplus::async::task<std::pair<int, std::string>>
```

Comment:
I feel we should move functions into a separate class and make it generic like AsyncCommandExecutor.

Also, please use RAII style to close FDs and Destroy fileaction object instead of manual.